packages/shylu/shylu_dd/bddc/src/shylu_SolverSuperLU.hpp

```diff
@@ -48,8 +48,8 @@
 #include <math.h>
 #include <string.h>
 
-#include "shylu_SolverBaseBDDC.h"
-#include "shylu_errorBDDC.h"
+#include "shylu_SolverBaseBDDC.hpp"
+#include "shylu_errorBDDC.hpp"
 
 typedef int int_t;
```

Comment:
It's a bit unfortunate to have a header put a typedef in the global namespace.  The macro `NO` just below this line is even worse.  Would this header file ever be exposed to users?  In general, I really wish we could hide SuperLU headers from users, as they tend to put all kinds of stuff in the global namespace.  Furthermore, I think hiding the `extern "C"` declarations behind wrappers would be a good idea.  We have this problem a lot with the BLAS, that Trilinos packages (e.g., kokkos-kernels) put `extern "C"` declarations in headers; if the application has its own `extern "C"` declarations, the two could conflict.

Reply:
I think the SuperLU header files should now be hidden from users. Riley was able to help me out on this yesterday. Also extern "C" declarations are in a .cpp file and the macro NO is gone.